apps/full-site-editing/full-site-editing-plugin/editor-site-launch/src/launch/index.tsx

```diff
@@ -57,11 +57,15 @@ const Launch: React.FunctionComponent< Props > = ( { onSubmit } ) => {
 
 	const CurrentLaunchStep = LaunchStepComponents[ currentStep ];
 
+	React.useEffect( () => {
+		if ( firstIncompleteStep && firstIncompleteStep !== LaunchStep.Name ) {
+			setStep( firstIncompleteStep );
+		}
+	}, [] );
```

Comment:
I wonder if this can be mitigated by introducing `getCurrentStep()` in the launch store. Not a blocker for this PR.

Reply:
Would the empty array fail ts lint check?

Reply:
> I wonder if this can be mitigated by introducing getCurrentStep() selector in the launch store. Not a blocker for this PR.
Check this comment that explains why this is done as a one time per component lifecycle thing:
https://github.com/Automattic/wp-calypso/pull/44882#issuecomment-673510881

> I wonder if this can be mitigated by introducing getCurrentStep() selector in the launch store. Not a blocker for this PR.

Just a warning. Empty array is how React recommends running some hook when the component is mounted.

Reply:
> This comment explains why this is done as a one time per component lifecycle thing. Once users start the flow we should leave them full control to navigate.

Great! Thanks for the explanation!